flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/catalog/FunctionCatalog.java

```diff
@@ -641,7 +641,8 @@ private void validateAndPrepareFunction(CatalogFunction function)
 
             FunctionDefinition definition = ((InlineCatalogFunction) function).getDefinition();
             UserDefinedFunctionHelper.prepareInstance(config, (UserDefinedFunction) definition);
-        } else if (function.getFunctionLanguage() == FunctionLanguage.JAVA) {
+        } else if (function.getFunctionLanguage() == FunctionLanguage.JAVA
+                && !(function instanceof InlineCatalogFunction)) {
```

Comment:
could you explain why add the code `!(function instanceof InlineCatalogFunction)`?
I guess you want to skip the validation?
But from the comments:
>  If the input is instance of `ScalarFunctionDefinition`, `TableFunctionDefinition` and so on, it means it uses the old type inference. We assume that they have been validated before being wrapped.
We may skip it when it's TableFunctionDefinition

Reply:
ok，It has been modified